src/lib/chips/chip-list.ts

```diff
@@ -126,35 +182,25 @@ export class MdChipList implements AfterContentInit {
     }
   }
 
-  /** Toggles the selected state of the currently focused chip. */
-  protected _toggleSelectOnFocusedChip(): void {
-    // Allow disabling of chip selection
-    if (!this.selectable) {
-      return;
-    }
-
-    let focusedIndex = this._keyManager.focusedItemIndex;
-
-    if (this._isValidIndex(focusedIndex)) {
-      let focusedChip: MdChip = this.chips.toArray()[focusedIndex];
-
-      if (focusedChip) {
-        focusedChip.toggleSelected();
-      }
-    }
-  }
-
   /**
    * Iterate through the list of chips and add them to our list of
    * subscribed chips.
    *
    * @param chips The list of chips to be subscribed.
    */
-  protected _subscribeChips(chips: QueryList<MdChip>): void {
+  protected _subscribeChips(chips: QueryList < MdChip >): void {
     chips.forEach(chip => this._addChip(chip));
   }
 
   /**
+   * Check the tab index as you should not be allowed to focus an empty list.
+   */
+  protected _checkTabIndex(): void {
+    // If we have 0 chips, we should not allow keyboard focus
+    this._tabIndex = (this.chips.length == 0 ? -1 : 0);
```

Comment:
Instead of manually calling `_checkTabIndex` and updating `_tabIndex`, why not just create a host binding like
```ts
host: {
  '[attr.tabindex]': 'chips.length ? 0 : -1'
}
```

However, this should also be overridden by any user-provided `tabIndex` value, so that makes it a bit more complex.